app/src/main/java/swati4star/createpdf/fragment/PPTtoPdfFragment.java

```diff
@@ -0,0 +1,212 @@
+import butterknife.OnClick;
+import swati4star.createpdf.R;
+import swati4star.createpdf.util.Constants;
+import swati4star.createpdf.util.FileUtils;
+import swati4star.createpdf.util.MorphButtonUtility;
+import swati4star.createpdf.util.PermissionsUtils;
+import swati4star.createpdf.util.RealPathUtil;
+import swati4star.createpdf.util.StringUtils;
+
+import static android.app.Activity.RESULT_OK;
+import static swati4star.createpdf.util.Constants.STORAGE_LOCATION;
+import static swati4star.createpdf.util.DialogUtils.createOverwriteDialog;
+import static swati4star.createpdf.util.StringUtils.getDefaultStorageLocation;
+import static swati4star.createpdf.util.StringUtils.getSnackbarwithAction;
+import static swati4star.createpdf.util.StringUtils.showSnackbar;
+
+public class PPTtoPdfFragment extends Fragment {
+
+    @BindView(R.id.create_PPTPdf)
+    MorphingButton mCreatePdf;
+    @BindView(R.id.tv_ppt_file_name)
+    TextView mTextView;
+
+    private Activity mActivity;
+    private FileUtils mFileUtils;
+    private Uri mPPTFileUri;
+    private String mRealPath;
+    private String mFileExtension;
+
+    private SharedPreferences mSharedPreferences;
+    private MorphButtonUtility mMorphButtonUtility;
+    private static final int PERMISSION_REQUEST_WRITE_EXTERNAL_STORAGE_RESULT = 1;
+    private boolean mPermissionGranted = false;
+    private boolean mButtonClicked = false;
+    private final int mFileSelectCode = 0;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View rootview = inflater.inflate(R.layout.fragment_pptto_pdf, container, false);
+        mSharedPreferences = PreferenceManager.getDefaultSharedPreferences(mActivity);
+        mMorphButtonUtility = new MorphButtonUtility(mActivity);
+        ButterKnife.bind(this, rootview);
+        mMorphButtonUtility.morphToGrey(mCreatePdf, mMorphButtonUtility.integer());
+        mCreatePdf.setEnabled(false);
+        return rootview;
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        mActivity = (Activity) context;
+        mFileUtils = new FileUtils(mActivity);
+    }
+
+    @OnClick(R.id.select_pptFile)
+    public void selectPPTFile() {
+        if (!mButtonClicked) {
+            Uri uri = Uri.parse(Environment.getRootDirectory() + "/");
+            Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
+            intent.setDataAndType(uri, "*/*");
+            intent.addCategory(Intent.CATEGORY_OPENABLE);
+            try {
+                startActivityForResult(
+                        Intent.createChooser(intent, String.valueOf(R.string.select_file)),
+                        mFileSelectCode);
+            } catch (android.content.ActivityNotFoundException ex) {
+                showSnackbar(mActivity, R.string.install_file_manager);
+            }
+            mButtonClicked = true;
```

Comment:
I think `mButtonClicked` should be set to true inside the try block. In case of any error, it should still remain as false otherwise the user won't be able to select ppt files again.

Reply:
Yes @srv-twry you are right, I have made the changes.